biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -397,6 +439,29 @@ public Resource getResource(String path) {
 		return resources.get(path);
 	}
 
+	/**
+	 * Returns a resource taking the release version into account.
+	 *
+	 * @param path the path of the resource to read
+	 * @param release the release to use
+	 * @return an optional representing the resource or an empty optional if the
+	 *         resource do not exits
+	 */
+	public Optional<Resource> getVersionedResource(String path, int release) {
```

Comment:
I think that this method isn't quite right or needs a different name and more documentation. At the moment it only checks one version of the release, not every version below, which isn't my understanding of how Multi-Release jars work.

If, for example, my jar has:

    /Foo.class
    /Bar.class
    /META-INF/versions/9/Foo.class
    /META-INF/versions/11/Bar.class

And I call:
*  `getVersionedResource(Foo.class, 11)` - I would expect to get returned the resource at `/META-INF/versions/9/Foo.class`, but this method will return an empty `Optional`
* `getVersionedResource(Bar.class, 9)` - I would expect to get returned the resource at `/Bar.class` but this method will return an empty `Optional`

Reply:
I can enhance the test-case but actually the method should work that way (but... if its not testes it most likely fails ..).

Reply:
That would be great - I'm definitely not infallible, and may simply be misunderstanding the role of the internal map structures.